src/test/java/com/amazon/ion/impl/IonReaderContinuableTopLevelBinaryTest.java

```diff
@@ -4436,4 +4448,190 @@ public void getAnnotationsAsStringFailsWhenTextIsUndefined(boolean constructFrom
+        // FlexSym then SID
+        "C9                       | Struct Length = 12 \n" +
+        "00                       | Switch to FlexSyms \n" +
+        "F9 6E 61 6D 65           | name \n" +
+        "5E                       | true \n" +
+        "0D                       | SID 6 \n " +
+        "5F                       | false",
+        // FlexSym field names using variable-length FlexSym type ID
+        "FD                       | Variable Length FlexSym struct \n" +
+        "1F                       | Length = 15 \n" +
+        "F9 6E 61 6D 65           | name \n" +
+        "5E                       | true \n" +
+        "F3 69 6D 70 6F 72 74 73  | imports \n " +
+        "5F                       | false",
+        // TODO delimited
+    })
+    public void readStruct_1_1(String inputBytes) throws Exception {
+        assertSimpleStructCorrectlyParsed(true, inputBytes);
+        assertSimpleStructCorrectlyParsed(false, inputBytes);
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void readMultipleNestedPrefixedStructs_1_1(boolean constructFromBytes) throws Exception {
+        byte[] input = hexStringToByteArray(cleanCommentedHexBytes(
+            "C8  | Struct Length = 8 \n" +
+            "17  | SID 11 \n" +
+            "C6  | Struct Length = 6 \n" +
+            "17  | SID 11 \n" +
+            "C4  | Struct Length = 4 \n" +
+            "17  | SID 11 \n" +
+            "C2  | Struct Length = 2 \n" +
+            "17  | SID 11 \n" +
+            "C0  | Struct Length = 0"
+        ));
+        reader = readerForIon11(input, constructFromBytes);
+        assertSequence(
+            container(IonType.STRUCT,
+                container(IonType.STRUCT,
+                    container(IonType.STRUCT)
+                )
+            ),
+            next(null)
+        );
+        closeAndCount();
+    }
+
+    /**
+     * Checks that the reader correctly reads symbol zero, which requires a special FlexSym, as a field name.
+     */
+    private void assertStructWithSymbolZeroFieldNamesCorrectlyParsed(boolean constructFromBytes, String inputBytes) throws Exception {
+        reader = readerForIon11(hexStringToByteArray(cleanCommentedHexBytes(inputBytes)), constructFromBytes);
+        assertSequence(
+            container(IonType.STRUCT,
+                next(IonType.BOOL), fieldNameSymbolZero(), booleanValue(true),
+                next(IonType.BOOL), fieldNameSymbolZero(), booleanValue(true),
+                next(IonType.BOOL), fieldName("name"), booleanValue(true),
+                next(IonType.BOOL), fieldNameSymbolZero(), booleanValue(true)
+            ),
+            next(null)
+        );
+        closeAndCount();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+        // SID 0 in fixed-length SID struct
+        "C8      | Struct Length = 8 \n" +
+        "01      | SID 0 \n" +
+        "5E      | true \n" +
+        "01      | FlexSym SID 0 \n" +
```

Comment:
This is a SID struct. This is not an attempt to switch to the FlexSym encoding, but to write symbol ID 0 as a SID.

Reply:
If FlexUInt 0 (`01`) is the flag to switch to a FlexSym struct, it can't also be SID 0. The only way to write SID 0 would be to switch to FlexSym struct and then write SID 0.